crates/walrus-sui/src/client/retry_client.rs

```diff
@@ -1190,26 +1195,23 @@ pub enum RetriableClientError {
     Other(#[from] anyhow::Error),
 }
 
-impl From<tokio::time::error::Elapsed> for RetriableClientError {
-    fn from(error: tokio::time::error::Elapsed) -> Self {
-        Self::RetryableTimeoutError(error)
-    }
-}
-
 impl From<tonic::Status> for RetriableClientError {
     fn from(status: tonic::Status) -> Self {
-        Self::RpcError(CheckpointRpcError::from(status))
+        Self::RpcError(CheckpointRpcError::from((status, 0)))
     }
 }
 
 impl RetriableClientError {
-    fn is_eligible_for_fallback(&self, next_checkpoint: u64) -> bool {
+    fn is_eligible_for_fallback(&self, next_checkpoint: u64, window_size: u64) -> bool {
         match self {
             Self::RpcError(rpc_error) if rpc_error.status.code() == tonic::Code::NotFound => {
-                rpc_error
-                    .status
-                    .checkpoint_height()
-                    .is_some_and(|height| next_checkpoint < height)
+                rpc_error.status.checkpoint_height().is_some_and(|height| {
+                    if next_checkpoint <= height {
+                        true
+                    } else {
+                        height + window_size >= next_checkpoint
```

Comment:
nit: can you add some documentation to why we need to consider window_size here? The important aspect is that this is a changing window based on the last height

Reply:
done